tt_eager/tt_dnn/op_library/prod/prod_op_all.cpp

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "tt_dnn/op_library/prod/prod_op_all.hpp"
+#include "tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp"
+
+#include <algorithm>
+#include <optional>
+
+#include "tt_metal/common/constants.hpp"
+#include <tt_eager/tt_numpy/functions.hpp>
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/tools/profiler/op_profiler.hpp"
+
+namespace tt {
+using namespace constants;
+namespace operations {
+namespace primary {
+
+void Prod_op::validate(const std::vector<Tensor>& input_tensors) const {
+    const auto& input_tensor_a = input_tensors.at(0);
+    TT_FATAL(input_tensor_a.storage_type() == StorageType::DEVICE, "Operands to eltwise unary need to be on device!");
+    TT_FATAL(input_tensor_a.buffer() != nullptr , "Operands to eltwise unary need to be allocated in buffers on device!");
+    TT_FATAL((input_tensor_a.get_layout() == Layout::TILE), "Inputs to eltwise unary must be tilized");
+    TT_FATAL(input_tensor_a.get_dtype() == DataType::BFLOAT16);
```

Comment:
Fix description (not eltwise unary). Also should be asserting for interleaved input and output mem config